File: yalo-challenge-java/src/main/java/com/yalo/challenge/logic/LogicExpressionProcessor.java
package com.yalo.challenge.logic;

import com.yalo.challenge.arithmetic.ArithmeticLexer;
import com.yalo.challenge.arithmetic.ArithmeticParser;
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.TokenStream;
import org.antlr.v4.runtime.tree.ParseTree;

import java.util.Map;

public class LogicExpressionProcessor {

    public Boolean process(String expression, Map<String, Double> context) {
        final CharStream input = CharStreams.fromString(expression);
        final ArithmeticLexer lexer = new ArithmeticLexer(input);
        final TokenStream tokenStream = new CommonTokenStream(lexer);
        final ArithmeticParser parser = new ArithmeticParser(tokenStream);
        final ParseTree expressionTree = parser.logicExpression();
        final LogicExpressionVisitor arithmeticVisitor = new LogicExpressionVisitor(context);
        return arithmeticVisitor.visit(expressionTree);
    }
}

File: yalo-challenge-node/lib/LogicExpressionProcessor.js
const antlr4 = require('antlr4');
const ArithmeticLexer = require('./grammar/ArithmeticLexer');
const ArithmeticParser = require('./grammar/ArithmeticParser');
const LogicExpressionVisitor = require('./LogicExpressionVisitor');


class LogicExpressionProcessor {
    process(expression, context) {
        let chars = new antlr4.InputStream(expression);
        let lexer = new ArithmeticLexer.ArithmeticLexer(chars);
        let tokens = new antlr4.CommonTokenStream(lexer);
        let parser = new ArithmeticParser.ArithmeticParser(tokens);
        let tree = parser.logicExpression();
        let visitor = new LogicExpressionVisitor(context);
        return visitor.visit(tree);
    }
}

module.exports = LogicExpressionProcessor;